packages/astro/src/assets/utils/transformToPath.ts

```diff
@@ -16,9 +16,19 @@ export function propsToFilename(transform: ImageTransform, hash: string) {
 	return `/${filename}_${hash}${outputExt}`;
 }
 
-export function hashTransform(transform: ImageTransform, imageService: string) {
+export function hashTransform(
+	transform: ImageTransform,
+	imageService: string,
+	propertiesToHash: string[]
+) {
 	// Extract the fields we want to hash
-	const { alt, class: className, style, widths, densities, ...rest } = transform;
-	const hashFields = { ...rest, imageService };
+	const hashFields = propertiesToHash.reduce(
+		(acc, prop) => {
+			// It's possible for `transform[prop]` here to be undefined, or null, but that's fine because it's still consistent
```

Comment:
nit: consistent ... to what exactly? Sorry, if I am going to read this comment new few weeks, I might not know what that means

Reply:
Sure, no worries. I actually had a longer message there that I removed because it overexplaining things a bit. I'll add more info

(for information, it's consistent between different transforms, as in, every transform that does not have a `height` explicitly set will end up with a `height: undefined` on it.)